src/common/snippets/include/snippets/lowered/loop_info.hpp

```diff
@@ -65,9 +60,9 @@ class LoopInfo {
     const std::vector<LoopPort>& get_exit_points() const;
     /**
      * @brief Returns handlers of loop specific iterations
-     * @return m_handlers
+     * @return handlers
      */
-    const SpecificIterationHandlers& get_handlers() const;
+    virtual const SpecificIterationHandlers& get_handlers() const = 0;
```

Comment:
You mentioned in the PR description: **_"The base class LoopInfo has many methods that aren't supported by ExpandedLoopInfo. It's needed for now to avoid many using get_loop_Info<UnifiedLoopInfo> in whole code base of library"_**
But one of the main goals of `LoopInfo` separation -- to split the interface and have in each class only the methods which are needed, right? And the methods which contain `LoopInfo` but are not supported by `ExpandedLoopInfo` are confusing me. It would seem that this PR is the right place to insert `get_loop_Info<UnifiedLoopInfo>` throughout the code base, if we want to get the clear interface of each class 😃. Or maybe I missed/forgot something?

@a-sidorova @IvanNovoselov I would propose discussing it offline